ui/lib/kv/addon/components/kv-delete-modal.js

```diff
@@ -0,0 +1,134 @@
+      case 'destroy':
+        return `This action will permanently destroy Version ${this.args.secret.version}
+        of the secret, and the secret data cannot be read or recovered later.`;
+      case 'delete metadata':
+        return 'This will permanently delete the metadata and versions of the secret. All version history will be removed. This cannot be undone.';
+      default:
+        return assert('mode must be one of undelete, delete, destroy, delete metadata.');
+    }
+  }
+
+  get generateDeleteRadioOptions() {
+    let isDeactivated = false;
+    if (this.args.secret.canReadMetadata) {
+      isDeactivated = this.args.metadata?.currentSecret.isDeactivated;
+    }
+
+    return [
+      {
+        key: 'delete-version',
+        label: 'Delete this version',
+        description: `This deletes Version ${this.args.secret.version} of the secret.`,
+        disabled: !this.args.secret.canDeleteVersion,
+        tooltipMessage: 'You do not have permission to delete a specific version.',
+      },
+      {
+        key: 'delete-latest-version',
+        label: 'Delete latest version',
+        description: 'This deletes the most recent version of the secret.',
+        disabled: !this.args.secret.canDeleteLatestVersion || isDeactivated,
+        tooltipMessage: isDeactivated
+          ? `The latest version of the secret is already ${this.args.metadata.secretState.state}.`
+          : 'You do not have permission to delete the latest version of this secret.',
+      },
+    ];
+  }
+
+  @action handleButtonClick(mode) {
+    // undelete is the only use case were a modal is not shown.
+    if (mode === 'undelete') {
+      this.deleteType = 'undelete-version';
+      this.save.perform();
+      return;
+    }
+    this.modalOpen = true;
+    // deleteType is the param used in the case/switch for deleteRecord on the adapter.
+    if (mode === 'destroy') {
+      this.deleteType = 'destroy-version';
+    }
+    if (mode === 'delete metadata') {
+      this.deleteType = 'delete-metadata';
+    }
+    // if mode is delete, they still need to select what kind of delete operation they'd like to perform so we don't set the deleteType until they select a radio option.
+  }
+
+  flashMessageMessage(isSuccess) {
+    if (isSuccess) {
+      if (this.deleteType === 'delete-metadata') {
+        return `Successfully deleted the metadata and all version data for the secret ${this.args.path}.`;
+      } else {
+        return `Successfully ${verbToPastTense(this.args.mode, 'past')} Version ${
+          this.args.secret.version
+        } of ${this.args.path}.`;
+      }
+    } else {
+      if (this.deleteType === 'delete-metadata') {
+        return `There was an issue deleting ${this.args.path} metadata.`;
+      } else {
+        return `There was an issue ${verbToPastTense(this.args.mode, 'gerund')} Version ${
+          this.args.secret.version
+        } of ${this.args.path}.`;
+      }
+    }
+  }
```

Comment:
i think `verbToPastTense` could instead be a const in this function: 

```suggestion
  
  message(isSuccess) {
    // if this.args.mode is adapter key

    const [verb] = this.args.mode.split('-');
    const message =
      this.deleteType === 'delete-metadata'
        ? `the metadata and all version data for the secret ${this.args.path}.`
        : `Version ${this.args.secret.version} of ${this.args.path}.`;
    if (isSuccess) {
      return `Successfully ${verb}ed ${message}`;
    } else {
      return `There was an issue ${verb}ing ${this.args.path} ${message}.`;
    }
  }
```

Reply:
actually this won't work and still might be hard to read. I think just a getter or something returning the word we need is best here